rust/datafusion/src/execution/context.rs

```diff
@@ -286,8 +288,20 @@ impl ExecutionContext {
                     .iter()
                     .map(|e| self.create_aggregate_expr(e, &input_schema))
                     .collect::<Result<Vec<_>>>()?;
+
+                let initial_aggr =
+                    HashAggregateExec::try_new(group_expr, aggr_expr, input)?;
+
+                let schema = initial_aggr.schema();
+                let partitions = initial_aggr.partitions()?;
+                let (final_group, final_aggr) = initial_aggr.make_final_expr();
+
+                let merge = Arc::new(MergeExec::new(schema.clone(), partitions));
```

Comment:
we could omit the merge stage entirely if the `initial_aggr` only has a single partition?

Reply:
This is a good point, I have updated it to return the initial_aggr when there is only a single partition, thanks!